### PR TITLE
Filter Widgets for event_event_type and event_event

### PR DESCRIPTION
**Pull request by gauravmittal1995**

### CODE REVIEW

File: modules/s3db/event.py [hunk 1]
```diff
@@ -265,33 +272,43 @@ def model(self):
         # Which levels of Hierarchy are we using?
         levels = current.gis.get_relevant_hierarchy_levels()
 
+        filter_widgets = [S3TextFilter(["name",
+                                        ],
+                                       label = T("Search"),
+                                       )
+                          ]
+
+        fw = filter_widgets
```
Comment (nursix) [inline modules/s3db/event.py]: What the point with this additional variable?

File: modules/s3db/event.py [hunk 2]
```diff
@@ -265,33 +272,43 @@ def model(self):
...
+                                    levels = levels,
+                                    label = T("Location"),
+                                    hidden = True,
+                                    ),
+                   # @ToDo: Filter for any event which starts or ends within a date range
+                   S3DateFilter("start_date",
+                                label = T("Date"),
+                                hide_time = True,
+                                hidden = True,
+                                input_labels = {"ge": "From", "le": "To"}
```
Comment (nursix) [inline modules/s3db/event.py]: That's the default

File: modules/s3db/event.py [hunk 3]
```diff
@@ -265,33 +272,43 @@ def model(self):
...
+
+        fw = filter_widgets
+
         if hierarchical_event_types:
-            filter_widgets = [S3HierarchyFilter("event_type_id",
-                                                label = T("Type"),
-                                                #multiple = False,
-                                                ),
-                              ]
+            fw += [S3HierarchyFilter("event_type_id",
```
Comment (nursix) [inline modules/s3db/event.py]: Using append() is faster than +=

File: modules/s3db/event.py (hunk 2 above)
Comment (gauravmittal1995) [inline modules/s3db/event.py]: Sorry, i dont understand ... what is default?

Comment (nursix) [inline modules/s3db/event.py]: These input labels.

File: modules/s3db/event.py [hunk 4]
```diff
@@ -265,37 +272,48 @@ def model(self):
...
 
         filter_widgets.extend((S3LocationFilter("event_location.location_id",
                                                 levels = levels,
                                                 label = T("Location"),
+                                                hidden = True,
                                                 ),
                                # @ToDo: Filter for any event which starts or ends within a date range
                                S3DateFilter("start_date",
-                                            label = None,
+                                            label = T("Date"),
```
Comment (gauravmittal1995) [inline modules/s3db/event.py]: @nursix I see, So should i remove the whole label option, or leave it as T("Date") ??

Comment (nursix) [inline modules/s3db/event.py]: Huh? No - I said these input_labels are default, so they do not need to be specified.

**Comment by gauravmittal1995** [comment]
@nursix  I have a question:
what of this todo? https://github.com/flavour/eden/blob/master/modules/s3db/pr.py#L3193#L3197
Should there be a autofill widget??

File: modules/s3db/event.py (hunk 4 above)
Comment (gauravmittal1995) [inline modules/s3db/event.py]: @nursix , Oh sorry, Didnt realise that! That was already there in the code, so i though u were reviewing the part which i changed ... will remove the input_labels

**Comment by gauravmittal1995** [comment]
@nursix Removed the input_label, Please Review

File: modules/s3db/event.py (hunk 4 above)
Comment (gauravmittal1995) [inline modules/s3db/event.py]: @nursix Done

**Pull request closed without merging**